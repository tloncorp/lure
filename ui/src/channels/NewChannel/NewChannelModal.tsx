import React from 'react';
import Dialog, { DialogContent } from '@/components/Dialog';
import { useDismissNavigate } from '@/logic/routing';
import NewChannelForm from './NewChannelForm';

export default function NewChannelModal() {
  const dismiss = useDismissNavigate();

  const onOpenChange = (isOpen: boolean) => {
    if (!isOpen) {
      dismiss();
    }
  };

  return (
    <Dialog defaultOpen onOpenChange={onOpenChange}>
      <DialogContent containerClass="w-full sm:max-w-lg">
        <NewChannelForm />
      </DialogContent>
    </Dialog>
  );
}
