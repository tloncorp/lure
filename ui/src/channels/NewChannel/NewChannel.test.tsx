/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { describe, expect, it } from 'vitest';
import { render } from '../../../test/utils';
import NewChannelModal from './NewChannelModal';

describe('NewChannelModal', () => {
  it('renders as expected', () => {
    const { asFragment } = render(<NewChannelModal />);
    expect(asFragment()).toMatchSnapshot();
  });
});
