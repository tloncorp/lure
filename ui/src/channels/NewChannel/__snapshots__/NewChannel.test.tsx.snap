// Vitest Snapshot v1

exports[`NewChannelModal > renders as expected 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    class="fixed inset-0 z-40 transform-gpu bg-black opacity-30"
    data-aria-hidden="true"
    data-state="open"
    style="pointer-events: auto;"
  />
  <section
    aria-describedby="radix-2"
    aria-labelledby="radix-1"
    class="dialog-container w-full sm:max-w-lg"
    data-state="open"
    id="radix-0"
    role="dialog"
    style="pointer-events: auto;"
    tabindex="-1"
  >
    <div
      class="dialog"
    >
      <div
        class="sm:w-96"
      >
        <header
          class="mb-3 flex items-center"
        >
          <h2
            class="text-lg font-bold"
          >
            Add New Channel
          </h2>
        </header>
      </div>
      <form
        class="flex flex-col"
      >
        <ul
          class="flex flex-col space-y-2 mb-5"
        >
          <li>
            <label
              class="flex cursor-pointer items-center justify-between rounded-lg p-2 bg-blue-soft"
            >
              <div
                class="flex w-full flex-col"
              >
                <div
                  class="flex flex-row items-center space-x-2"
                >
                  <div
                    class="rounded p-2 mix-blend-multiply dark:mix-blend-screen bg-blue-soft"
                  >
                    <svg
                      class="h-6 w-6 text-gray-600"
                      fill="none"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        class="fill-current"
                        d="m4.44444 14.636-.94419.3294.94419-.3294Zm14.68296 1.001-.9806.1961-.0683-.3414.1585-.31.8904.4553Zm.7255 3.6276.9806-.1962-.9806.1962Zm-.5883.5883-.1962.9806.1962-.9806Zm-3.6276-.7255-.4553-.8904.31-.1585.3414.0683-.1961.9806Zm-6.27299.4282.32938-.9442-.32938.9442ZM5 12c0 .8099.13714 1.5857.38864 2.3066l-1.88839.6588C3.17587 14.0355 3 13.0372 3 12h2Zm7-7c-3.86599 0-7 3.13401-7 7H3c0-4.97056 4.02944-9 9-9v2Zm7 7c0-3.86599-3.134-7-7-7V3c4.9706 0 9 4.02944 9 9h-2Zm-.763 3.1817c.4875-.9535.763-2.034.763-3.1817h2c0 1.4716-.3539 2.8634-.9822 4.0922l-1.7808-.9105Zm1.871.2591.7255 3.6276-1.9612.3923-.7255-3.6276 1.9612-.3923Zm.7255 3.6276c.2099 1.0496-.7155 1.975-1.7651 1.7651l.3923-1.9612c-.3499-.0699-.6583.2385-.5884.5884l1.9612-.3923Zm-1.7651 1.7651-3.6276-.7255.3923-1.9612 3.6276.7255-.3923 1.9612ZM12 19c1.1477 0 2.2282-.2755 3.1817-.763l.9105 1.7808C14.8634 20.6461 13.4716 21 12 21v-2Zm-2.30661-.3886C10.4143 18.8629 11.1901 19 12 19v2c-1.0372 0-2.03554-.1759-2.96538-.5002l.65877-1.8884Zm-4.30475-4.3048c.70113 2.0098 2.29492 3.6036 4.30475 4.3048l-.65877 1.8884c-2.58577-.9021-4.63233-2.9487-5.53437-5.5344l1.88839-.6588Z"
                      />
                    </svg>
                  </div>
                  <div
                    class="flex w-full flex-col justify-start text-left"
                  >
                    <span
                      class="font-semibold"
                    >
                      Chat
                    </span>
                    <span
                      class="text-sm font-medium text-gray-600"
                    >
                      A simple, standard text chat
                    </span>
                  </div>
                </div>
              </div>
              <input
                class="sr-only"
                name="type"
                type="radio"
                value="chat"
              />
            </label>
          </li>
          <li>
            <label
              class="flex cursor-pointer items-center justify-between rounded-lg p-2 bg-white"
            >
              <div
                class="flex w-full flex-col"
              >
                <div
                  class="flex flex-row items-center space-x-2"
                >
                  <div
                    class="rounded p-2 mix-blend-multiply dark:mix-blend-screen bg-green-soft"
                  >
                    <svg
                      class="h-6 w-6 text-gray-600"
                      fill="none"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        class="fill-current"
                        clip-rule="evenodd"
                        d="M5 3C3.89543 3 3 3.89543 3 5V13C3 14.1046 3.89543 15 5 15H8.01894C8.27426 18.3562 11.0784 21 14.5 21C18.0899 21 21 18.0899 21 14.5C21 11.0784 18.3562 8.27426 15 8.01894V5C15 3.89543 14.1046 3 13 3H5ZM13 8.17393V5L5 5V13H8.17393C8.73735 10.6148 10.6148 8.73735 13 8.17393ZM19 14.5C19 16.9853 16.9853 19 14.5 19C12.0147 19 10 16.9853 10 14.5C10 12.0147 12.0147 10 14.5 10C16.9853 10 19 12.0147 19 14.5Z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </div>
                  <div
                    class="flex w-full flex-col justify-start text-left"
                  >
                    <span
                      class="font-semibold"
                    >
                      Gallery
                    </span>
                    <span
                      class="text-sm font-medium text-gray-600"
                    >
                      Gather, connect, and arrange rich media
                    </span>
                  </div>
                </div>
              </div>
              <input
                class="sr-only"
                name="type"
                type="radio"
                value="heap"
              />
            </label>
          </li>
          <li>
            <label
              class="flex cursor-pointer items-center justify-between rounded-lg p-2 bg-white"
            >
              <div
                class="flex w-full flex-col"
              >
                <div
                  class="flex flex-row items-center space-x-2"
                >
                  <div
                    class="rounded p-2 mix-blend-multiply dark:mix-blend-screen bg-red-soft"
                  >
                    <svg
                      class="h-6 w-6 text-gray-600"
                      fill="none"
                      viewBox="0 0 24 24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        class="fill-current"
                        d="M4 4V3H3V4H4ZM4 19H3V20H4V19ZM7 22C7 22.5523 7.44772 23 8 23C8.55228 23 9 22.5523 9 22H7ZM11.5 8.5C10.9477 8.5 10.5 8.94772 10.5 9.5C10.5 10.0523 10.9477 10.5 11.5 10.5V8.5ZM15.5 10.5C16.0523 10.5 16.5 10.0523 16.5 9.5C16.5 8.94772 16.0523 8.5 15.5 8.5V10.5ZM3 4V19H5V4H3ZM18 6V17H20V6H18ZM4 5H8V3H4V5ZM8 5H17V3H8V5ZM17 18H8V20H17V18ZM8 18H4V20H8V18ZM7 4V19H9V4H7ZM7 19V22H9V19H7ZM11.5 10.5H15.5V8.5H11.5V10.5ZM18 17C18 17.5523 17.5523 18 17 18V20C18.6569 20 20 18.6569 20 17H18ZM20 6C20 4.34315 18.6569 3 17 3V5C17.5523 5 18 5.44772 18 6H20Z"
                      />
                    </svg>
                  </div>
                  <div
                    class="flex w-full flex-col justify-start text-left"
                  >
                    <span
                      class="font-semibold"
                    >
                      Notebook
                    </span>
                    <span
                      class="text-sm font-medium text-gray-600"
                    >
                      Longform publishing and discussion
                    </span>
                  </div>
                </div>
              </div>
              <input
                class="sr-only"
                name="type"
                type="radio"
                value="diary"
              />
            </label>
          </li>
        </ul>
        <label
          class="mb-3 font-semibold"
        >
          Channel Name*
          <input
            class="input my-2 block w-full p-1"
            name="meta.title"
            type="text"
          />
        </label>
        <label
          class="mb-3 font-semibold"
        >
          Channel Permissions
          <ul
            class="flex flex-col space-y-2"
          >
            <li>
              <label
                class="flex cursor-pointer items-center justify-between space-x-2 py-2"
              >
                <div
                  class="flex items-center"
                >
                  <div
                    class="h-4 w-4 rounded-xl border-4 border-gray-400"
                  />
                </div>
                <div
                  class="flex w-full flex-col"
                >
                  <div
                    class="flex flex-row items-center space-x-2"
                  >
                    <div
                      class="flex w-full flex-col justify-start text-left"
                    >
                      <span
                        class="font-semibold"
                      >
                        Open to All
                      </span>
                      <span
                        class="text-sm font-medium text-gray-600"
                      >
                        Anyone can view and write
                      </span>
                    </div>
                  </div>
                </div>
                <input
                  class="sr-only"
                  name="privacy"
                  type="radio"
                  value="public"
                />
              </label>
            </li>
            <li>
              <label
                class="flex cursor-pointer items-center justify-between space-x-2 py-2"
              >
                <div
                  class="flex items-center"
                >
                  <div
                    class="h-4 w-4 rounded-xl border-2 border-gray-200"
                  />
                </div>
                <div
                  class="flex w-full flex-col"
                >
                  <div
                    class="flex flex-row items-center space-x-2"
                  >
                    <div
                      class="flex w-full flex-col justify-start text-left"
                    >
                      <span
                        class="font-semibold"
                      >
                        Members Can View
                      </span>
                      <span
                        class="text-sm font-medium text-gray-600"
                      >
                        Members can only view
                      </span>
                    </div>
                  </div>
                </div>
                <input
                  class="sr-only"
                  name="privacy"
                  type="radio"
                  value="read-only"
                />
              </label>
            </li>
            <li>
              <label
                class="flex cursor-pointer items-center justify-between space-x-2 py-2"
              >
                <div
                  class="flex items-center"
                >
                  <div
                    class="h-4 w-4 rounded-xl border-2 border-gray-200"
                  />
                </div>
                <div
                  class="flex w-full flex-col"
                >
                  <div
                    class="flex flex-row items-center space-x-2"
                  >
                    <div
                      class="flex w-full flex-col justify-start text-left"
                    >
                      <span
                        class="font-semibold"
                      >
                        Admin Only
                      </span>
                      <span
                        class="text-sm font-medium text-gray-600"
                      >
                        Only Admin can view and write
                      </span>
                    </div>
                  </div>
                </div>
                <input
                  class="sr-only"
                  name="privacy"
                  type="radio"
                  value="secret"
                />
              </label>
            </li>
          </ul>
        </label>
        <label
          class="flex cursor-pointer items-center justify-between space-x-2 py-2"
        >
          <div
            class="flex items-center"
          >
            <div
              class="h-4 w-4 rounded-sm border-2 border-gray-200"
            />
          </div>
          <div
            class="flex w-full flex-col"
          >
            <div
              class="flex flex-row items-center space-x-2"
            >
              <div
                class="flex w-full flex-col justify-start text-left"
              >
                <span
                  class="font-semibold"
                >
                  Default
                </span>
                <span
                  class="text-sm font-medium text-gray-600"
                >
                  New members will be automatically subscribed to this channel
                </span>
              </div>
            </div>
          </div>
          <input
            class="sr-only"
            name="join"
            type="checkbox"
          />
        </label>
        <footer
          class="mt-4 flex items-center justify-between space-x-2"
        >
          <div
            class="ml-auto flex items-center space-x-2"
          >
            <button
              class="secondary-button ml-auto"
              type="button"
            >
              Cancel
            </button>
            <button
              class="button"
              disabled=""
              type="submit"
            >
              Add Channel
            </button>
          </div>
        </footer>
      </form>
      <button
        class="icon-button absolute top-6 right-6"
        type="button"
      >
        <svg
          class="h-4 w-4"
          fill="none"
          viewBox="0 0 16 16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            class="stroke-current"
            d="m8 8 4-4M8 8 4 4m4 4-4 4m4-4 4 4"
            stroke-linecap="round"
            stroke-width="2"
          />
        </svg>
      </button>
    </div>
  </section>
</DocumentFragment>
`;
